checker/src/org/checkerframework/checker/upperbound/UpperBoundStore.java

```diff
@@ -0,0 +1,157 @@
+            AnnotationMirror newAnno =
+                    factory.createLTLengthOfAnnotation(UpperBoundUtils.getValue(anno));
+            CFValue val =
+                    analysis.createSingleAnnotationValue(
+                            factory.getQualifierHierarchy().leastUpperBound(newAnno, anno),
+                            rec.getType());
+            replace.put(rec, val);
+        } else if (AnnotationUtils.containsSameByClass(atm, LTLengthOf.class)) {
+            AnnotationMirror anno = AnnotationUtils.getAnnotationByClass(atm, LTLengthOf.class);
+            AnnotationMirror newAnno =
+                    factory.createLTEqLengthOfAnnotation(UpperBoundUtils.getValue(anno));
+            CFValue val =
+                    analysis.createSingleAnnotationValue(
+                            factory.getQualifierHierarchy().leastUpperBound(newAnno, anno),
+                            rec.getType());
+            replace.put(rec, val);
+        } else if (AnnotationUtils.containsSameByClass(atm, LTEqLengthOf.class)) {
+            CFValue val = analysis.createSingleAnnotationValue(factory.UNKNOWN, rec.getType());
+            replace.put(rec, val);
+        }
+    }
+
+    private void applyAdd(
+            Receiver rec,
+            Map<Receiver, CFValue> replace,
+            AnnotatedTypeFactory atypeFactory,
+            Receiver caller) {
+
+        UpperBoundAnnotatedTypeFactory factory = (UpperBoundAnnotatedTypeFactory) atypeFactory;
+        CFValue value = this.getValue(rec);
+        Set<AnnotationMirror> atm = value.getAnnotations();
+        if (AnnotationUtils.containsSameByClass(atm, LTEqLengthOf.class)) {
+            AnnotationMirror anno = AnnotationUtils.getAnnotationByClass(atm, LTEqLengthOf.class);
+            String[] vals = UpperBoundUtils.getValue(anno);
+            if (vals.length != 1 || !vals[0].equals(caller.toString())) {
+                return;
+            }
+            AnnotationMirror newAnno = factory.createLTLengthOfAnnotation(caller.toString());
+            CFValue val =
+                    analysis.createSingleAnnotationValue(
+                            factory.getQualifierHierarchy().greatestLowerBound(newAnno, anno),
+                            rec.getType());
+            replace.put(rec, val);
+        } else if (AnnotationUtils.containsSameByClass(atm, LTLengthOf.class)) {
+            AnnotationMirror anno = AnnotationUtils.getAnnotationByClass(atm, LTLengthOf.class);
+            String[] vals = UpperBoundUtils.getValue(anno);
+            if (vals.length != 1 || !vals[0].equals(caller.toString())) {
+                return;
+            }
+            AnnotationMirror newAnno = factory.createLTOMLengthOfAnnotation(caller.toString());
+            CFValue val =
+                    analysis.createSingleAnnotationValue(
+                            factory.getQualifierHierarchy().greatestLowerBound(newAnno, anno),
```

Comment:
In this case it's definitely wrong - we know that the type's value is only a single name of an array, so there's no need. I'll remove it from this method.